bundles/org.dataflowanalysis.analysis/src/org/dataflowanalysis/analysis/core/AbstractVertex.java

```diff
@@ -126,40 +126,86 @@ public boolean isSource() {
     /**
      * Returns a list of node characteristic with the given characteristic type
      * <p>
-     * See {@link AbstractVertex#getDataFlowVariablesWithName(String)} for a similar method for data flow
+     * See {@link AbstractVertex#getDataFlowVariables(String)} for a similar method for data flow
      * variables
      * @param characteristicType Name of the characteristic type
      * @return Returns a list of all node characteristics matching the characteristic type
      */
-    public List<CharacteristicValue> getNodeCharacteristicsWithName(String characteristicType) {
+    public List<CharacteristicValue> getNodeCharacteristics(String characteristicType) {
         return this.getAllNodeCharacteristics().stream()
                 .filter(cv -> cv.getTypeName().equals(characteristicType))
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Returns a list of node characteristic value names with the given characteristic type
+     * <p>
+     * See {@link AbstractVertex#getDataFlowVariableNames(String)} for a similar method for data flow
+     * variables
+     * @param characteristicType Name of the characteristic type
+     * @return Returns a list of all node characteristics matching the characteristic type
+     */
+    public List<String> getNodeCharacteristicNames(String characteristicType) {
+        return this.getAllNodeCharacteristics().stream()
+                .filter(cv -> cv.getTypeName().equals(characteristicType))
+                .map(CharacteristicValue::getValueName)
+                .collect(Collectors.toList());
+    }
+
     /**
      * Returns a list of data flow variables with the given name
      * <p>
-     * See {@link AbstractVertex#getNodeCharacteristicsWithName(String)} for a similar method for node characteristics
+     * See {@link AbstractVertex#getNodeCharacteristics(String)} for a similar method for node characteristics
```

Comment:
Just a minor thing: Should this be called Vertex Characteristics?

Reply:
Fixed in 75c539e86348d740484e6fd54e82efb28cfd36a3